sky/server/server.py

```diff
@@ -267,6 +273,112 @@ async def dispatch(self, request: fastapi.Request, call_next):
+            return fastapi.responses.JSONResponse(
+                status_code=401,
+                content={
+                    'detail': 'Bearer token required for service account access'
+                })
+
+        # Extract and validate service account token
+        sa_token = auth_header.split(' ', 1)[1]
+        if not sa_token.startswith('sky_'):
+            return fastapi.responses.JSONResponse(
+                status_code=401,
+                content={'detail': 'Invalid service account token format'})
+
+        try:
+            # Import here to avoid circular imports
+            # pylint: disable=import-outside-toplevel
+            from sky.users.token_service import token_service
+
+            # Verify and decode JWT token
+            payload = token_service.verify_token(sa_token)
+
+            if payload is None:
+                logger.warning(
+                    'BearerTokenMiddleware: Token verification failed')
+                return fastapi.responses.JSONResponse(
+                    status_code=401,
+                    content={
+                        'detail': 'Invalid or expired service account token'
+                    })
+
+            # Extract user information from JWT payload
+            user_id = payload.get('sub')  # Subject = service account user ID
+            user_name = payload.get('name')
+            token_id = payload.get('token_id')
+
+            if not user_id or not token_id:
+                logger.warning(
+                    'Invalid token payload: missing user_id or token_id')
+                return fastapi.responses.JSONResponse(
+                    status_code=401,
+                    content={'detail': 'Invalid token payload'})
+
+            # Verify user still exists in database
+            user_info = global_user_state.get_user(user_id)
+            if user_info is None:
+                logger.warning(f'Token user {user_id} no longer exists')
+                return fastapi.responses.JSONResponse(
+                    status_code=401,
+                    content={'detail': 'Service account user no longer exists'})
+
+            # Update last used timestamp for token tracking
+            try:
+                global_user_state.update_service_account_token_last_used(
+                    token_id)
+            except Exception as e:  # pylint: disable=broad-except
+                # Don't fail authentication if we can't update last used time
+                logger.debug(f'Failed to update token last used time: {e}')
+
+                # Set the authenticated user - this integrates with Casbin RBAC
+            auth_user = models.User(id=user_id,
+                                    name=user_name or user_info.name)
+            request.state.auth_user = auth_user
+
+            # Override user info in request body for service account requests
+            await _override_user_info_in_request_body(request, auth_user)
+
+            logger.info(
```

Comment:
```suggestion
            logger.debug(
```

Info feels too verbose for middleware